hedera-node/src/main/java/com/hedera/services/grpc/NettyGrpcServerManager.java

```diff
@@ -73,14 +81,29 @@ public void start(int port, int tlsPort, Consumer<String> println) {
 		}
 	}
 
-	private Server startOneNettyServer(boolean sslEnabled, int port, Consumer<String> println) throws Exception {
+	Server startOneNettyServer(boolean sslEnabled, int port, Consumer<String> println, Pause pause) throws Exception {
 		println.accept(nettyAction("Starting", sslEnabled, port, true));
 
 		NettyServerBuilder builder = nettyBuilder.builderFor(port, sslEnabled);
 		bindableServices.forEach(builder::addService);
 		serviceDefinitions.forEach(builder::addService);
 		Server server = builder.build();
-		server.start();
+
+		var retryNo = 1;
+		final var n = Math.max(0, nodeProperties.nettyStartRetries());
+		for (; retryNo <= n; retryNo++) {
+			try {
+				server.start();
+				break;
+			} catch (IOException e) {
+				final var summaryMsg = nettyAction("Still trying to start", sslEnabled, port, true);
+				log.warn("(Attempts=" + retryNo + ") " + summaryMsg + e.getMessage());
+				pause.forMs(nodeProperties.nettyStartRetryIntervalMs());
+			}
+		}
+		if (retryNo == n + 1) {
+			server.start();
+		}
```

Comment:
Can this `if` statement be removed if line 94 is `final var n = Math.max(1, nodeProperties.nettyStartRetries());`?

Reply:
It could, but then there would need to be an `if` inside the `catch` to log with the same pattern... 😁

Reply:
Hmm, I don't see why an `if` is needed in the `catch` block. My understanding is if `server.start()` inside the `for` is successful, nothing will happen and the `break` will end the loop and move on. If it's not successful, the `catch` block will log the attempt times, pause, and then re-run `server.start()` till the `server.start()` finishes successful or the max retry number is reached.